lib/src/components/NxFieldset/types.ts

```diff
@@ -23,3 +23,29 @@ export const propTypes: PropTypes.ValidationMap<Props> = {
   validationErrors: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.string.isRequired), PropTypes.string]),
   isPristine: PropTypes.bool
 };
+
+// The following are for the state helpers
+export type RadioValidator = (value: string | null) => ValidationErrors;
+export type CheckboxValidator = (values: string[]) => ValidationErrors;
+export type RadioSetter = (v: string | null) => void;
+export type CheckboxState = [boolean, () => void];
+export type CheckboxInitValues<K extends string | number> = Record<K, boolean>;
+export type CheckboxStates<K extends string | number> = Record<K, CheckboxState>;
+
```

Comment:
Just for my understanding, is type CheckboxStates essentially an object, in which each of it's keys are type string or number, and the values associated with each key are type checkboxState (and therefore an array containing a boolean and a function- in this case the toggle function)?

Reply:
> Just for my understanding, is type CheckboxStates essentially an object, in which each of it's keys are type string or number, and the values associated with each key are type checkboxState (and therefore an array containing a boolean and a function- in this case the toggle function)?

Yes, exactly.

Reply:
> > Just for my understanding, is type CheckboxStates essentially an object, in which each of it's keys are type string or number, and the values associated with each key are type checkboxState (and therefore an array containing a boolean and a function- in this case the toggle function)?
>
> Yes, exactly.

Oh I wanted to elaborate on this a little.  It's not quite that `CheckboxState`'s keys are strings or numbers, it's that they can be specific strings and numbers, known at compile time.  So if it's known at compile time that it's an object like `{ foo: ..., bar: ... }` (say, because it was written as an object literal), then `K` becomes just `'foo' | 'bar'`, not `string | number`.  In this way the compiler can track what keys are involved and what keys will be present in objects derived from this object.

Reply:
> Oh I wanted to elaborate on this a little. It's not quite that CheckboxState's keys are strings or numbers, it's that they can be specific strings and numbers, known at compile time. So if it's known at compile time that it's an object like { foo: ..., bar: ... } (say, because it was written as an object literal), then K becomes just 'foo' | 'bar', not string | number. In this way the compiler can track what keys are involved and what keys will be present in objects derived from this object.

Sorry, I never saw this comment. Thanks for expanding on this a bit more.